### PR TITLE
Reverting Batchnorm implementation vector processing changes

### PR DESCRIPTION
**Pull request by muralinr**
Reverting Batchnorm vector processing changes

### CODE REVIEW

File: src/kernels/MIOpenBatchNormBwdSpatial.cl [hunk 1]
```diff
@@ -293,7 +293,7 @@ MIOpenBatchNormBwdSpatial(const __global _FLOAT* __restrict x_in,
 #if(MIO_BN_USESAVED == 0)
     //==== CALC MEAN and VARIANCE ONCE AGAIN =======================
     _FLOAT_PREC variance = (_FLOAT_PREC)0.;
-#if(MIO_BN_HW >= 4096)
+#if(0)
```
Comment (JehandadKhan) [inline src/kernels/MIOpenBatchNormBwdSpatial.cl:296]: @muralinr  Please follow the workaround protocol to document this effort and ensure that we are able to re-enable it again in the future with a proper fix.

**Review by JehandadKhan — CHANGES_REQUESTED** [review]

File: src/kernels/MIOpenBatchNormBwdSpatial.cl (hunk 1 above)
Comment (JehandadKhan) [reply src/kernels/MIOpenBatchNormBwdSpatial.cl:296]: [Here is a link](https://github.com/ROCmSoftwarePlatform/MIOpen/issues/561)

**Comment by atamazov** [comment]
@muralinr Please clearly explain why we need this, and I will be able to suggest on implementation.

File: src/kernels/MIOpenBatchNormBwdSpatial.cl [hunk 2]
```diff
@@ -429,7 +429,7 @@ MIOpenBatchNormBwdSpatial(const __global _FLOAT* __restrict x_in,
     nidx                = remkey / MIO_BN_HW;
     hwidx               = remkey - (nidx * MIO_BN_HW);
     index               = nidx * MIO_BN_CHW + chwid + hwidx;
-    if(index < MIO_BN_NCHW)
+    if(index < (MIO_BN_NCHW - 3))
```
Comment (atamazov) [inline src/kernels/MIOpenBatchNormBwdSpatial.cl:432]: It is possible that MIO_BN_NCHW <= 3 here?

Comment (muralinr) [reply src/kernels/MIOpenBatchNormBwdSpatial.cl:432]: We are reading 4 inputs at a time.  Here index is going beyond MIO_BN_NCHW and hence I put it to (MIO_BN_NCHW -3 ) to cap it at  MIO_BN_NCHW.

Comment (muralinr) [reply src/kernels/MIOpenBatchNormBwdSpatial.cl:432]: We are reading 4 inputs at a time and indexing is going beyond MIO_BN_NCHW.  I am limiting at (MIO_BN_NCHW -3) to cap it at MIO_BN_NCHW.

Comment (atamazov) [reply src/kernels/MIOpenBatchNormBwdSpatial.cl:432]: The answer is "No"?

File: src/kernels/MIOpenBatchNormBwdSpatial.cl (hunk 1 above)
Comment (atamazov) [reply src/kernels/MIOpenBatchNormBwdSpatial.cl]: @muralinr Are you going to quickly make necessary changes? If not, then I will do.

Comment (muralinr) [reply src/kernels/MIOpenBatchNormBwdSpatial.cl]: Hi Artem, I am coming up with proper fix without workaround.  I pushed changes. I am working with framework teams to test these changes..

File: src/kernels/MIOpenBatchNormBwdSpatial.cl (hunk 2 above)
Comment (muralinr) [reply src/kernels/MIOpenBatchNormBwdSpatial.cl:432]: yes Artem

**Comment by atamazov** [comment]
Converted to draft to avoid spoiling docker image cache on CI. Please continue working normally. I will change this back to "normal" PR myself.

**Review by atamazov — APPROVED** [review]
LGTM

**Review by JehandadKhan — APPROVED** [review]

**Review by atamazov — DISMISSED** [review]
LGTM

**Review by atamazov — DISMISSED** [review]

**Review by atamazov — APPROVED** [review]

**Pull request merged**